src/idl_gen_cpp.cpp

```diff
@@ -2507,7 +2506,8 @@ class CppGenerator : public BaseGenerator {
           }
         } else {
           const auto ptype = GenTypeNativePtr(
-              NativeName(name, type.struct_def, opts_), &afield, true);
+              WrapNativeNameInNameSpace(*type.struct_def, opts_), &afield,
```

Comment:
Probably declaration of the `name` at L2496 can be moved to a line just before L2505.

Reply:
Good point. Done.